katsdpcal/test/test_control.py

```diff
@@ -914,6 +914,90 @@ async def test_capture_complex_models(self):
         # but calibration is performed using the full sky model.
         await self.test_capture(expected_BG_rtol=5e-2, expected_BCROSS_DIODE_rtol=1e-2,
                                 expected_K_rtol=3e-3)
+        
+    async def test_reference_pointing_capture(self):
+        """Tests the reference pointing feature with some data, and checks that 
+        solutions are computed and checks dtypes and shapes.
+        """
+        # Number of dumps spent tracking, slewing and awaiting
+        n_track = 16
+        n_slew = 8
+        n_await = 3
+        # Number of pointings
+        n_pointing = 8
+        n_times = n_track + n_slew + n_await
+        rs = np.random.RandomState(seed=1)
+        
+        # Create target with pointingcal tag
+        target = ('J1331+3030, radec pointingcal, '
+                  '13:31:08.29, +30:30:33.0, (0 50e3 0.1823 1.4757 -0.4739 0.0336)')
+        self.telstate.add('cbf_target', target, ts=0.003)
+        target = katpoint.Target(self.telstate.cbf_target)
+        telstate_cb = self.telstate.view('cb')
+        # Adding track, track, slew obs activities for each offset
+        n_activity=0
+        for pointing in range(0,n_pointing):
+            telstate_cb.add('obs_activity', 'track',
+                            ts=self.first_dump_ts + (n_activity - 0.5) * self.dump_period)
+            telstate_cb.add('obs_activity', 'slew',
+                            ts=self.first_dump_ts + ((n_activity + 2) - 0.5) * self.dump_period)
+            n_activity+=3
+
+        telstate_cb.add('obs_activity', 'await_pipeline',
+                                ts=self.first_dump_ts + ((n_track+n_slew) - 0.5) * self.dump_period)
+        
+        # Creating antenna objects
+        ants = [self.telstate[a+"_observer"] for a in self.antennas]
+        ant_objects = [katpoint.Antenna(a) for a in ants]
+        
+        max_extent = 1.0
+        # Build up sequence of pointing offsets running linearly in x and y directions
+        scan = np.linspace(-max_extent, max_extent, n_pointing// 2)
+        offsets_along_x = np.c_[scan, np.zeros_like(scan)]
+        offsets_along_y = np.c_[np.zeros_like(scan), scan]
+        offsets = np.r_[offsets_along_y, offsets_along_x]
+
+        # Updating pos_actual_scan_azim/elev every 0.5 seconds (4 times per dump)
+        pos_sensor_period = 0.5
+        num_pos_updates = self.dump_period / pos_sensor_period
+        start_time = self.first_dump_ts- (0.5* self.dump_period)
+        for ant in ant_objects:
+            ts = start_time
+            for offset in offsets:
+                for i in range(num_pos_updates * (n_track + n_slew) / n_pointing):
```

Comment:
The answer should be an int and not a float...
```suggestion
                for i in range(num_pos_updates * (n_track + n_slew) // n_pointing):
```